pkg/kine/drivers/generic/generic.go

```diff
@@ -394,15 +396,32 @@ func (d *Generic) queryRowPrepared(ctx context.Context, txName, sql string, prep
 }
 
 func (d *Generic) InitializeWriteBatching(ctx context.Context) {
-	if d.BatchingEnabled && d.BatchingInterval > 0 && d.BatchingMaxQueries > 0 {
-		d.flushCh = make(chan struct{})
-		d.addToBatchQueueCh = make(chan BatchedInsert)
-		d.removeFromBatchQueueCh = make(chan int)
-		d.batchingQueue = make([]BatchedInsert, 0, d.BatchingMaxQueries)
 
-		go d.watchBatchQueue(ctx)
-		go d.batchQueueWatcher(ctx)
+	if !d.BatchingEnabled {
+		return
 	}
+
+	if d.BatchingInterval < 10*time.Millisecond {
+		logrus.WithFields(logrus.Fields{"interval": d.BatchingInterval, "queue_size": d.BatchingMaxQueries}).Print("Interval is less than 10ms, disabling batching of insert queries")
+		return
+	}
+
+	if d.BatchingMaxQueries < 5 {
```

Comment:
We also need an upper limit - up until ~120 works fine for me. After that I get 
```
time="2023-10-20T16:14:45+02:00" level=error msg="Process batch queue error" error="too many SQL variables"
``` errors